### ISSUE TITLE
Generic.WhiteSpace.ScopeIndent error when using array destructuring with exact indent checking

### ISSUE THREAD

**Issue by dingo-d** [squizlabs/PHP_CodeSniffer#2479]
I'm having issues with this sniff. If I don't use `exact` property (which I've read I shouldn't in the customisation wiki), then I may miss on over indented code (and I wouldn't like this). If I use it, the errors for over indentation will happen (yay), but then I get errors reported in destructured arrays inside class methods 

```php
class TestClass {
  public function test() {
    $options = [
      'enabled'     => true,
      'compression' => 'gzip',
    ];

    [
      'enabled'     => $enabled,
      'compression' => $compression,
    ] = $options;
  }
}
```

This will report issues on lines 9 and 10, telling me to indent them 4 instead of 6 spaces (I'm using 2 space indent without tabs).

When I write this outside class, I don't get any errors. Also just assigning an array to a variable doesn't cause any errors (in or out of class).

If I don't use `exact`the error isn't shown, because I 'should' indent them 4 spaces, but they are indented 6 so it's not exact and it's ok (and it's not).

Is this a bug?

**Comment by gsherwood** [comment]
> Is this a bug?

Yes, I think there is something wrong here. This sniff disables exact indenting between arrays, but it's not working in this specific case. I think I know why though, so just looking into it.

**Comment by gsherwood** [comment]
This is now fixed and will be in the next release. Thanks for reporting it.

**Comment by dingo-d** [comment]
Awesome! Thanks!
